web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -165,27 +188,9 @@ BmcMangaList.prototype.generateComic = function(comic) {
     comicSrcList.classList.toggle('nested');
     elm.appendChild(comicSrcList);
 
+    let self = this;
     comic.iterSources(source => {
-        const srcElem = document.createElement('div');
-        srcElem.classList.add('label-container');
-
-        const srcLink = document.createElement('div');
-        srcLink.classList.add('label');
-        srcLink.innerText = source.reader;
-        srcLink.bmcData = {
-            reader: source.reader,
-            name: source.name,
-        };
-        srcLink.onclick = this.onSourceClick.bind(this, comic, source);
-        srcElem.appendChild(srcLink);
-
-        const deleteSrcIcon = document.createElement('span');
-        deleteSrcIcon.classList.add('fa', 'fa-trash');
-        deleteSrcIcon.setAttribute('aria-hidden', 'true');
-        deleteSrcIcon.onclick = this.onSourceDelete.bind(this);
-        srcElem.appendChild(deleteSrcIcon);
-
-        comicSrcList.appendChild(srcElem);
+        self.generateSource(comic, source, comicSrcList);
```

Comment:
why binding this to self ?

Reply:
I have a trauma with `this` being reassigned in callbacks.